src/bun.js/bindings/KeyObject.cpp

```diff
@@ -0,0 +1,2352 @@
+                            alg = CryptoAlgorithmIdentifier::ECDSA;
+                            result = CryptoKeyEC::platformImportSpki(CryptoAlgorithmIdentifier::ECDSA, curve, Vector<uint8_t>((uint8_t*)data, byteLength), true, CryptoKeyUsageVerify);
+                        }
+                        if (result == nullptr) {
+                            auto scope = DECLARE_THROW_SCOPE(vm);
+                            throwException(globalObject, scope, createTypeError(globalObject, "Invalid EC public key"_s));
+                            return JSValue::encode(JSC::jsUndefined());
+                        }
+                        auto impl = result.releaseNonNull();
+                        return JSC::JSValue::encode(JSCryptoKey::create(structure, zigGlobalObject, WTFMove(impl)));
+                    } else {
+                        auto scope = DECLARE_THROW_SCOPE(vm);
+                        throwException(globalObject, scope, createTypeError(globalObject, "Unsupported public key"_s));
+                        return JSValue::encode(JSC::jsUndefined());
+                    }
+                }
+
+                auto scope = DECLARE_THROW_SCOPE(vm);
+                JSC::throwTypeError(globalObject, scope, "type should be 'pkcs1' or 'spki'"_s);
+                return JSValue::encode(JSC::jsUndefined());
+            }
+            auto scope = DECLARE_THROW_SCOPE(vm);
+            JSC::throwTypeError(globalObject, scope, "format should be 'pem' or 'der'"_s);
+            return JSValue::encode(JSC::jsUndefined());
+        }
+        auto scope = DECLARE_THROW_SCOPE(vm);
+        JSC::throwTypeError(globalObject, scope, "expected options to be a object"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+    auto scope = DECLARE_THROW_SCOPE(vm);
+    JSC::throwTypeError(globalObject, scope, "expected options to be a object"_s);
+    return JSC::JSValue::encode(JSC::JSValue {});
+}
+
+JSC::EncodedJSValue KeyObject__createSecretKey(JSC::JSGlobalObject* lexicalGlobalObject, JSC::CallFrame* callFrame)
+{
+
+    JSValue bufferArg = callFrame->uncheckedArgument(0);
+
+    if (bufferArg.isCell()) {
+        auto type = bufferArg.asCell()->type();
+
+        switch (type) {
+
+        case DataViewType:
+        case Uint8ArrayType:
+        case Uint8ClampedArrayType:
+        case Uint16ArrayType:
+        case Uint32ArrayType:
+        case Int8ArrayType:
+        case Int16ArrayType:
+        case Int32ArrayType:
+        case Float32ArrayType:
+        case Float64ArrayType:
+        case BigInt64ArrayType:
+        case BigUint64ArrayType: {
+            JSC::JSArrayBufferView* view = jsCast<JSC::JSArrayBufferView*>(bufferArg.asCell());
+
+            void* data = view->vector();
+            size_t byteLength = view->length();
+            if (UNLIKELY(!data)) {
+                break;
+            }
+            Zig::GlobalObject* globalObject = reinterpret_cast<Zig::GlobalObject*>(lexicalGlobalObject);
```

Comment:
This needs a test to see what happens if it's called from inside a `node:vm`. If it is a `JSGlobalObject`, it will segfault.